pkg/controllers/clusterset/clusterdeployment/clusterdeployment_controller.go

```diff
@@ -113,25 +154,41 @@ func (r *Reconciler) Reconcile(ctx context.Context, req ctrl.Request) (ctrl.Resu
 		}
 		return ctrl.Result{}, err
 	}
-
-	//if the clusterdeployment is not created by clusterpool or already claimed, do nothing
-	if clusterdeployment.Spec.ClusterPoolRef == nil || len(clusterdeployment.Spec.ClusterPoolRef.ClaimName) != 0 {
-		return ctrl.Result{}, nil
-	}
-
-	clusterpool := &hivev1.ClusterPool{}
-	err = r.client.Get(ctx, types.NamespacedName{Namespace: clusterdeployment.Spec.ClusterPoolRef.Namespace, Name: clusterdeployment.Spec.ClusterPoolRef.PoolName}, clusterpool)
-	if err != nil {
-		if errors.IsNotFound(err) {
-			return ctrl.Result{}, nil
+	var targetLabels map[string]string
+	//if the clusterdeployment is not created by clusterpool, update label follow managedcluster label
+	if clusterdeployment.Spec.ClusterPoolRef == nil {
+		//get managedclusterlabel
+		managedcluster := clusterv1.ManagedCluster{}
+		err := r.client.Get(ctx,
+			types.NamespacedName{
+				Name: clusterdeployment.Namespace,
+			},
+			&managedcluster,
+		)
+		if err != nil {
+			if errors.IsNotFound(err) {
+				return ctrl.Result{}, nil
+			}
+			return ctrl.Result{}, err
 		}
-		return ctrl.Result{}, err
+		targetLabels = managedcluster.Labels
```

Comment:
sync all labels or just set label?

Reply:
it's just set label. 
In the blow func，we only sync the set filed
`utils.SyncMapFiled(&isModified, &clusterdeployment.Labels, targetLabels, **clustersetutils.ClusterSetLabel**)`